src/components/expression/expression-loader.js

```diff
@@ -24,101 +28,249 @@ genotet.utils.inherit(genotet.ExpressionLoader, genotet.ViewLoader);
+          genotet.warning('input gene not found');
+          return;
+        }
+        if (data.allConditionNames.length == 0) {
+          genotet.warning('input condition not found');
+          return;
+        }
+
+        this.data.matrixAll = data;
+        this.signal('allMatrixDataLoaded');
+      }.bind(this), 'jsonp')
+      .fail(this.fail.bind(this, 'cannot load expression matrix', params));
   };
 
-  $.get(genotet.data.serverURL, params, function(data) {
-      // Store the last applied data selectors.
-      _.extend(data, {
-        matrixname: matrixName,
-        geneRegex: geneRegex,
-        conditionRegex: conditionRegex
-      });
+/**
+ * Implements the expression matrix loading ajax call. Since the matrix may
+ * contain a large number of entries, we use GET request.
+ * @param {string} matrixName Name of the expression matrix.
+ * @param {string} dataName Name of the expression matrix data file.
+ * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} conditionNames Names for experiment condition
+ *      selection.
+ * @private
+ */
+genotet.ExpressionLoader.prototype.loadExpressionMatrix_ =
+  function(matrixName, dataName, geneNames, conditionNames) {
+    var params = {
+      type: 'expression',
+      matrixName: matrixName,
+      geneNames: geneNames,
+      conditionNames: conditionNames
+    };
 
-      this.data.matrix = data;
+    $.get(genotet.data.serverURL, params, function(data) {
+        // Store the last applied data selectors.
+        _.extend(data, {
+          matrixname: matrixName,
+          dataName: 'b-subtilis'
+        });
 
-      this.signal('loadComplete');
-    }.bind(this), 'jsonp')
-    .fail(this.fail.bind(this, 'cannot load expression matrix', params));
-};
+        if (data.geneNames.length == 0) {
+          genotet.warning('input gene not found');
+          return;
+        }
+        if (data.conditionNames.length == 0) {
+          genotet.warning('input condition not found');
+          return;
+        }
+
+        this.signal('loadStart');
+        this.data.matrix = data;
+
+        this.loadTfaData_(matrixName, dataName, geneNames, conditionNames);
+      }.bind(this), 'jsonp')
+      .fail(this.fail.bind(this, 'cannot load expression matrix', params));
+  };
+
+/**
+ * Implements the TFA data loading ajax call.
+ * Since the matrix may contain a large number of entries, we use GET request.
+ * @param {string} matrixName Name of the expression matrix.
+ * @param {string} dataName Name of the expression matrix data file.
+ * @param {!Array<string>} geneNames Names for gene selection.
+ * @param {!Array<string>} conditionNames Names for experiment condition
+ *      selection.
+ * @private
+ */
+genotet.ExpressionLoader.prototype.loadTfaData_ =
+  function(matrixName, dataName, geneNames, conditionNames) {
+    var tfaParams = {
+      type: 'expression-profile',
+      matrixName: 'b-subtilis',
```

Comment:
However, the expression create UI only have dataName selection. Where is the matrixName selection?